exercises/knapsack/.meta/src/reference/java/Knapsack.java

```diff
@@ -0,0 +1,201 @@
+import java.util.ArrayList;
+
+/**
+ * Knapsack item with a weight and value
+ */
+class Item {
+
+    // Weight of the item
+    int weight;
+    // Value of the item
+    int value;
+
+    Item(int itemWeight, int itemValue) {
+        this.weight = itemWeight;
+        this.value = itemValue;
+    }
+    
+}
+
+class Knapsack {
```

Comment:
Concerning the algorithm you used:
I think it is too complicated and hard to follow (maybe this will improve after removing comments).
I checked for other algorithms solving this well known coding problem, and, for example, I like [this article](https://medium.com/@ssaurel/solving-the-knapsack-problem-in-java-c985c71a7e64) and the solution presented.

Reply:
I've added comments so that it would easy to follow. I've coded three approaches

- Plain recursion
- Top down optimization
- Bottom up optimization

I'm not sure if it would be good, if I directly write the bottom up approach. If you feel the other way round, I'll remove all other code and keep the bottom up approach in the reference implementation.

Reply:
I see. But for the reference implementation, you only need to provide one algorithm that solves the problem. Which one would you use?

Reply:
As Bottom up is the best one, I may use that if only one algorithm is supposed to be there.

Reply:
Cleaned the file